### PR TITLE
[Feature] Add Affinity to GroupSpec

### PR DESCRIPTION
**Pull request by ajanikow**

### CODE REVIEW

File: pkg/deployment/deployment_affinity_test.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+		required, role)
+
+	for _, mod := range mods {
+		mod(affinity)
+	}
+
+	return affinity
+}
+
+func TestEnsurePod_ArangoDB_Affinity(t *testing.T) {
```
Comment (ewoutp) [inline pkg/deployment/deployment_affinity_test.go]: TestEnsurePod_ArangoDB_Affinity  -> TestEnsurePod_ArangoDB_AntiAffinity ??

File: pkg/apis/deployment/v1/server_group_spec.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ type ServerGroupSpec struct {
 	VolumeClaimTemplate *v1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
 	// VolumeResizeMode specified resize mode for pvc
 	VolumeResizeMode *PVCResizeMode `json:"pvcResizeMode,omitempty"`
+	// AntiAffinity specified additional antiAffinity settings in ArangoDB Pod definitions
```
Comment (ewoutp) [inline pkg/apis/deployment/v1/server_group_spec.go]: Should there also be Affinity?

Comment (ajanikow) [reply pkg/apis/deployment/v1/server_group_spec.go]: I can add also Affinity and Node Affinity  - so it will cover whole functionality

Comment (ajanikow) [reply pkg/apis/deployment/v1/server_group_spec.go]: Done

Comment (ajanikow) [reply pkg/apis/deployment/v1/server_group_spec.go]: Done

**Pull request merged**